Backend/Interview.Domain/Events/Events/RoomEvent.cs

```diff
@@ -49,7 +50,7 @@ public RoomEvent(Guid roomId, string type, T? value, bool stateful, Guid created
     {
     }
 
-    public string? BuildStringPayload()
+    public string? BuildStringPayload(IRoomEventSerializer serializer)
```

Comment:
Why doesn't the serializer encapsulate in the class (constructor)?

Reply:
Yes, because it's not necessary in all cases.